libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -660,6 +803,32 @@ TEST( MQTT_Unit_API, OperationWaitTimeout )
                                                              0 );
         TEST_ASSERT_NOT_NULL( _pMqttConnection );
 
+
+        /* Getting the free index from the MQTT connection to MQTT context mapping array. */
+        contextIndex = _IotMqtt_getFreeIndexFromContextConnectionArray();
+
+        /* Creating Mutex for the synchronization of MQTT Context used for sending the packets
+         * on the network using MQTT LTS API. */
+        contextMutex = IotMutex_CreateRecursiveMutex( &( connToContext[ contextIndex ].contextMutex ) );
+
+        /* Create the subscription mutex for a new connection. */
+
+        if( contextMutex == true )
+        {
+            /* Assigning the MQTT Connection. */
+            connToContext[ contextIndex ].mqttConnection = _pMqttConnection;
+
+            subscriptionMutexCreated = IotMutex_CreateNonRecursiveMutex( &( connToContext[ contextIndex ].subscriptionMutex ) );
+
+            if( subscriptionMutexCreated == false )
+            {
+                IotLogError( "Failed to create subscription mutex for new connection." );
+            }
+
+            /* Initializing the MQTT context used in calling MQTT LTS API. */
+            managedMqttStatus = MQTT_Init( &( connToContext[ contextIndex ].context ), &transport, &callbacks, &networkBuffer );
+        }
```

Comment:
A lot of the code seems to be the same as _setContext function? can that be reused here?

Reply:
Addressed